CRM/Case/Form/Case.php

```diff
@@ -177,22 +201,52 @@ public function preProcess() {
     }
     $this->assign('clientName', isset($this->_currentlyViewedContactId) ? $contact->display_name : NULL);
 
-    $session = CRM_Core_Session::singleton();
-    $this->_currentUserId = $session->get('userID');
+    $this->_currentUserId = CRM_Core_Session::getLoggedInContactID();
 
-    //Add activity custom data is included in this page
-    CRM_Custom_Form_CustomData::preProcess($this, NULL, $this->_activityTypeId, 1, 'Activity');
     $className = "CRM_Case_Form_Activity_{$this->_activityTypeFile}";
-    $className::preProcess($this);
-    $activityGroupTree = $this->_groupTree;
+    switch ($className) {
+      // @todo flesh out this switch. Sure we hate switches but it's better than
+      // than the default below for code tracing purposes.
+      case 'CRM_Case_Form_Activity_OpenCase':
```

Comment:
It's fine like this but not sure if you're aware this form is only ever used to create a new case, and to delete/restore cases, so it either returns early or this is always set to OpenCase. I don't remember the history but this is either copy/paste from activity, or the other way around.

Reply:
@demeritcowboy no - definitely not aware - in that case we maybe make it clearer in the code

Reply:
@demeritcowboy so this? 
![image](https://github.com/civicrm/civicrm-core/assets/336308/9524545d-be0a-494f-aef6-c2ea9c00639b)

Reply:
that works too